pkg/cri/server/image_pull_test.go

```diff
@@ -443,3 +443,34 @@ func TestSnapshotterFromPodSandboxConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestImageGetLabels(t *testing.T) {
+	sandboxImage := "k8s.gcr.io/pause"
+	criService := newTestCRIService()
+	criService.config.SandboxImage = sandboxImage
```

Comment:
The test case makes me think about if `k8s.gcr.io/pause:lastest` equals `k8s.gcr.io/pause`.
- as I know, the config loader will load the sandboxImage name directly.
- the crictl will use `distribution.ParseDockerRef` to parse the image

Should we parse the `criService.config.SandboxImage` here to align?

Reply:
@pacoxu you want to say we should handle the case when image tag is not specified in config.toml? like
`sandbox_image = "registry.k8s.io/pause"` will be equivalent to `sandbox_image="registry.k8s.io/pause:latest"`?

Reply:
there may also be use with the sha. Like `registry.k8s.io/pause:3.9@sha256:F00BA2...` or `registry.k8s.io/pause@sha256:F00BA2...`

Reply:
The `ref` used in `getLabels`, which is used to compare with `c.config.SandboxImage`, is derived from: 

```
imageRef := r.GetImage().GetImage()
namedRef, err := distribution.ParseDockerRef(imageRef)
ref := namedRef.String()
```

And the `r` in the code above is `c.config.SandboxImage` [1]. So I think the problem here boils down to:

if `distribution.ParseDockerRef(c.config.SandboxImage) == c.config.SandboxImage` then we are good, otherwise we are not.

Based on the comment on `ParseDockerRef` [2], the returned ref can only be either tagged or digested, but not both. So if we are using `registry.k8s.io/pause:3.9` or `registry.k8s.io/pause@sha256:F00BA2...` then we are good. But if we are using `registry.k8s.io/pause:3.9@sha256:F00BA2...` then the comparison in `getLabels` will return false and the image will not be pinned.

[1] https://github.com/containerd/containerd/blob/66f186d42dc804284699443bc4de5ed637215a75/pkg/cri/server/sandbox_run.go#L105

[2] https://github.com/containerd/containerd/blob/d2f68bfb36141a90d04fdcb5793d66d3b883bebf/reference/docker/normalize.go#L73-L79

Reply:
Agree. 

They may be some corner cases that we should take care here.

Reply:
Thank @pacoxu for pointing out and @ruiwen-zhao for another corner case and explanation. so we have two corner cases
1. neither tag nor  digest specified in config.toml
2.  tag and digest both specified in config.toml

Reply:
@pacoxu @ruiwen-zhao @SergeyKanzhelev updated to handle the corner cases

Reply:
yep parsing the sandbox image from config does the trick.: https://github.com/containerd/containerd/pull/7944/files#diff-127e86d81515f945d265d390de70eff677aeca5863736d904275a3b8f971011dR298

Thanks for changing this!